gateway-ha/src/main/java/io/trino/gateway/ha/router/ProcessedRequest.java

```diff
@@ -0,0 +1,524 @@
+                    log.warn("Could not deserialize bearer token as json");
+                }
+            }
+            return null;
+        });
+    }
+
+    private Optional<String> extractUser(HttpServletRequest request, String userField)
+    {
+        String header;
+        header = request.getHeader(TRINO_USER_HEADER_NAME);
+        if (header != null) {
+            return Optional.of(header);
+        }
+
+        Optional<String> user = extractUserFromAuthorizationHeader(request.getHeader("Authorization"), userField);
+        if (user.isPresent()) {
+            return user;
+        }
+
+        return extractUserFromCookies(request, userField);
+    }
+
+    public String getBody()
+    {
+        return body;
+    }
+
+    public String getQueryType()
+    {
+        return queryType;
+    }
+
+    public String getDefaultSchema()
+    {
+        return defaultSchema;
+    }
+
+    public Set<QualifiedName> getTables()
+    {
+        return tables;
+    }
+
+    public boolean tablesContains(String testName)
+    {
+        char dot = '.';
+        char quote = '"';
+        List<Identifier> parts = new ArrayList<>();
+        int start = 0;
+        boolean inQuotes = false;
+        boolean partQuoted = false;
+        for (int i = 0; i < testName.length(); i++) {
+            if (testName.charAt(i) == quote) {
+                if (!inQuotes) {
+                    if (i != start) {
+                        log.error("Illegal position for first quote character in table name: %s", testName);
+                        return false;
+                    }
+                    start = start + 1;
+                    partQuoted = true;
+                }
+                if (inQuotes && testName.charAt(i - 1) == '\\') {
+                    continue;
+                }
+
+                inQuotes = !inQuotes;
+                continue;
+            }
+            if (testName.charAt(i) == dot && !inQuotes) {
+                if (partQuoted) {
+                    parts.add(new Identifier(testName.substring(start, i - 1)));
+                }
+                else {
+                    parts.add(new Identifier(testName.substring(start, i)));
+                }
+                start = i + 1;
+                partQuoted = false;
+            }
+        }
+        if (partQuoted) {
+            parts.add(new Identifier(testName.substring(start, testName.length() - 1)));
+        }
+        else {
+            parts.add(new Identifier(testName.substring(start, testName.length())));
+        }
+
+        return tables.contains(QualifiedName.of(parts));
+    }
+
+    public String getDefaultCatalog()
+    {
+        return defaultCatalog;
+    }
+
+    public Set<String> getCatalogs()
+    {
+        return catalogs;
+    }
+
+    public Set<String> getSchemas()
+    {
+        return schemas;
+    }
+
+    public Set<String> getCatalogSchemas()
+    {
+        return catalogSchemas;
+    }
```

Comment:
Remove unused methods.

Reply:
These are intended for use in routing rules, but are unused in the Trino Gateway code. I added more cases to `routing_rules_processed_request.yml` to cover `getDefaultCatalog()` and `getCatalogSchemas()` and provide usage examples.

Reply:
@ebyhr , any feedback?

Reply:
Please add `@SuppressWarnings("unused")` annotation to unused methods.